File: backend/semgrep_rules/tests/python/kisa_sw_18_missing_authentication.py
from rest_framework.permissions import AllowAny, IsAuthenticated
from rest_framework.decorators import permission_classes

# ruleid: kisa.sw18.python.public-sensitive-endpoint-review
@permission_classes([AllowAny])
def unsafe(request):
    return "sensitive action"

# ok: kisa.sw18.python.public-sensitive-endpoint-review
@permission_classes([IsAuthenticated])
def safe(request):
    return "protected"

File: backend/semgrep_rules/tests/java/kisa_sw_18_missing_authentication.java
import jakarta.annotation.PermitAll;

class kisa_sw_18_missing_authentication {
    // ruleid: kisa.sw18.java.public-sensitive-endpoint-review
    @PermitAll
    public void unsafe() {
        System.out.println("sensitive action");
    }

    public void safe() {
        // ok: kisa.sw18.java.public-sensitive-endpoint-review
        System.out.println("protected elsewhere");
    }
}
